kubemarine/cri/containerd.py

```diff
@@ -212,19 +275,50 @@ def configure_crictl(group: NodeGroup) -> None:
     group.put(StringIO(crictl_config), '/etc/crictl.yaml', backup=True, sudo=True)
 
 
+def get_config_path(inventory: dict) -> Optional[str]:
+    config_path: Optional[str] = inventory.get('containerdConfig', {}) \
+        .get('plugins."io.containerd.grpc.v1.cri".registry', {}).get('config_path')
+    return config_path
+
+
 def configure_containerd(group: NodeGroup) -> RunnersGroupResult:
     cluster = group.cluster
     log = cluster.log
-    config_toml = get_containerd_config_as_toml(cluster)
+
+    # Dump containerd configuration
+    config_toml = get_config_as_toml(cluster.inventory.get("services", {}).get("cri", {}).get('containerdConfig', {}))
     config_string = toml.dumps(config_toml)
     utils.dump_file(cluster, config_string, 'containerd-config.toml')
+    config_path = config_toml['plugins']['io.containerd.grpc.v1.cri']['registry'].get('config_path')
+
+    # Dump registries configuration
+    registries_config = {}
+    for registry, host_config in cluster.inventory.get("services", {}).get("cri", {})\
+            .get('containerdRegistriesConfig', {}).items():
+        # sanitize_filepath is needed for windows/macOS, where some symbols are restricted in file_path,
+        # but they can be used in mirror name (e.g. ':')
+        registries_dir = sanitize_filepath(f'registries/{registry}', replacement_text='_')
+        os.makedirs(utils.get_dump_filepath(cluster.context, registries_dir))
+        registry_host_toml = get_config_as_toml(host_config)
+        registries_config[registry] = toml.dumps(registry_host_toml)
+        utils.dump_file(cluster, registries_config[registry], sanitize_filepath(f"{registries_dir}/hosts.toml"))
```

Comment:
Looks like here sanitize_filepath is redundant as utils.dump_file also does sanitizing

Reply:
fixed